filebeat/crawler/prospector.go

```diff
@@ -80,27 +82,29 @@ func (p *Prospector) Init() error {
 
 // Starts scanning through all the file paths and fetch the related files. Start a harvester for each file
 func (p *Prospector) Run(wg *sync.WaitGroup) {
-
-	// TODO: Defer the wg.Done() call to block shutdown
-	// Currently there are 2 cases where shutting down the prospector could be blocked:
-	// 1. reading from file
-	// 2. forwarding event to spooler
-	// As this is not implemented yet, no blocking on prospector shutdown is done.
-	wg.Done()
+	defer wg.Done()
```

Comment:
instead of forwarding wg to prospector can we have the wait-group handling in crawler. Reduces need to pass to many 'wg' variables + makes prospector unaware of fact it is run within a go-routine.

e.g. 

```
for i, p := range crawler.prospectors {
  c.wg.Add(1)
  go func(id int, prospector *Prospector){
    defer c.wg.Done()
    logp.Debug("starting prospector %v", id)
    defer logp.Debug("prospector %v stopped", id)
    prospector.Run()
  }(i, p)
}
```